sqlglot/optimizer/scope.py

```diff
@@ -140,6 +157,62 @@ def selects(self):
             return []
         return self.expression.selects
 
+    @property
+    def correlations(self):
+        """
+        Columns referenced by correlated subqueries.
+
+        Returns:
+            list[tuple[exp.Column, Scope]]: List of Column instances from
+                child scopes that reference this scope, along with the child
+                scope instance itself.
+        """
+        result = []
```

Comment:
should this be a dictionary?

Reply:
There might be instances multiple of the same logical column. And expressions hash on value, not identity.